libraries/rush-sdk/src/RushSdkLoader.ts

```diff
@@ -0,0 +1,248 @@
+            throw new Error(`The ${RUSH_LIB_NAME} package failed to load`);
+          }
+        }
+
+        if (RushSdkLoader._rushLibModule !== undefined) {
+          // to track which scenario is active and how it got initialized.
+          global.___rush___rushLibModuleFromInstallAndRunRush = RushSdkLoader._rushLibModule;
+          terminal.writeVerboseLine(`Loaded ${RUSH_LIB_NAME} installed by install-run-rush`);
+        }
+      } catch (e) {
+        // This error indicates that a project is trying to import "@rushstack/rush-sdk", but the Rush engine
+        // instance cannot be found.  If you are writing Jest tests for a Rush plugin, add "@microsoft/rush-lib"
+        // to the devDependencies for your project.
+        throw new Error(
+          'The @rushstack/rush-sdk package was not able to load the Rush engine:' + (e as Error).message
+        );
+      } finally {
+        RushSdkLoader._rushLibModuleHasBeenInstalled = true;
+      }
+
+      if (!RushSdkLoader._rushLibModule) {
+        throw new Error('The @rushstack/rush-sdk package was not able to load the Rush engine.');
+      }
+    }
+  }
+
+  private static _tryLoadRushLib(): _RushLibModuleType | undefined {
+    // SCENARIO 1:  Rush's PluginManager has initialized "rush-sdk" with Rush's own instance of rush-lib.
+    // The Rush host process will assign "global.___rush___rushLibModule" before loading the plugin.
+    let rushLibModule: _RushLibModuleType | undefined =
+      global.___rush___rushLibModule || global.___rush___rushLibModuleFromInstallAndRunRush;
+
+    // SCENARIO 2:  The project importing "rush-sdk" has installed its own instance of "rush-lib"
+    // as a package.json dependency.  For example, this is used by the Jest tests for Rush plugins.
+    if (rushLibModule === undefined) {
+      const thisProjectRoot: string | undefined =
+        PackageJsonLookup.instance.tryGetPackageFolderFor(__dirname);
+      if (!thisProjectRoot) {
+        throw new Error('Unable to find package.json for @rushstack/rush-sdk.');
+      }
+
+      let callingModule: NodeModule | null | undefined = module?.parent;
+      while (callingModule?.filename?.startsWith(thisProjectRoot)) {
+        // Walk up calling modules until we're out of this pacakge.
+        callingModule = callingModule.parent;
+      }
+
+      const importingPath: string | undefined = callingModule?.filename;
+      if (importingPath !== undefined) {
+        const callerPackageFolder: string | undefined =
+          PackageJsonLookup.instance.tryGetPackageFolderFor(importingPath);
+
+        if (callerPackageFolder !== undefined) {
+          const callerPackageJson: IPackageJson = RushSdkLoader._require(
+            `${callerPackageFolder}/package.json`
+          );
+
+          // Does the caller properly declare a dependency on rush-lib?
+          if (
+            (callerPackageJson.dependencies && callerPackageJson.dependencies[RUSH_LIB_NAME] !== undefined) ||
```

Comment:
```suggestion
            (callerPackageJson.devDependencies?.[RUSH_LIB_NAME] ??
              callerPackageJson.dependencies?.[RUSH_LIB_NAME] ??
              callerPackageJson.peerDependencies?.[RUSH_LIB_NAME] ??
              callerPackageJson.optionalDependencies?.[RUSH_LIB_NAME]) !== undefined
```

Reply:
Sorry, I don't know how to get GitHub to apply a multi-line suggested change. The other lines were meant to be replaced by this as well.

Reply:
... and now I've read how to do it.